### DISCUSSION TITLE
Trigger script on incoming SIP connection?

### DISCUSSION THREAD

**Discussion by timmrogers** [baresip/baresip#2823] [Q&A]
I have been successful in compiling the current version of baresip for Debian Linux 12 and configuring it as an IP PA. It is working nicely for both SIP extensions/page groups and for multicast.

However, some our locations are noisy, and I'd like to be able to trigger...something (flashing image on the screen; external device) when certain extensions are paged. I assume that this is possible - but I can't figure out how to do it! 

Is there a way to run a script on incoming SIP connections? What about incoming multicast messages?

Thank you!

**Comment by nonibytes** [comment]
You would want to write a simple C module to do this. The module should simplify listen for a connection event and trigger the script. If you don't know how to do this you can probably ask ChatGPT or Gemini AI.

Here's some information on modules:

https://github.com/baresip/baresip/wiki/How-to-add-new-modules
